### DISCUSSION TITLE
Better events intergration: Improved UI or Thunderbird and other client intergration?

### DISCUSSION THREAD

**Discussion by yuannan** [ALikesToCode/plasma-applet-eventcalendar#18] [Ideas]
I've personally started using thunderbird for work and find the calendar view to be much easier to use, track, and sync than the one built into event calendar. I personally would like better events management as I use my computer as my core task management system. I'm sure other would also like this, maybe to lesser extents, or cannot due to the lack of current features.

There are 3 main approaches that I think we can take:

### Improve the current events management:
Make the current events systems in EC better, it is currently unintuitive to set new events, it took me a while to figure out that I needed to click the sidebar with the date in order to add a new event. This should be more intuitive to the user, perhaps something as simple as an "add event" button at the bottom?
#### Pros:
- Less fragmentation in the short term, people can use an improved EC without much more work. Simply polishing an existing feature to be more refined.
- Can be merged to upstream.
#### Cons:
- More fragmentation in the long term. While dev work from upstream (Zren) has appeared to slowed down, work on plasma 6 does seem to be in progress. I'm uncertain of @ALikesToCode 's perspective on if we should consider this repo as a fork or as a working repo before a merge upstream. Many PRs upstream are still open and it remains to be seen if they ever will be merged. This is unfortunately an aspect of FOSS work as many of us (myself included) have other obligations, professional or otherwise, outside of voluntary FOSS passion work. This could lead to a nasty situation with diverging feature sets and fragmentation. Dividing efforts between 2 repos. FOSS work usually needs a certain level of community and momentum to keep going. I've worked on another plasmoid (Panon) before and have stopped contributing to the project as the mountain of work was huge. API calls all the way down from python, to c, to system level had to be changed to improve the actual project. While focusing on one repo is not optimal for other reasons, it does keep momentum and at least user support more streamlined. Issues and discussions were not added until recently to this repo for example, with myself not realising this fork existing until I finally got around to fixing my events despite upstream being broken for months. By further dividing our efforts between this repo and upstream we could kill what little momentum EC has left.
- On the subject of diverging feature sets if Zren decides to take upstream in a different direction or leave PRs open it could mean that this repo would be forced to be a fork. The majority of users still report PRs and issues upstream, it would take significant momentum to shift users over to this repo. In that instance we would be the new maintainers of EC.  I personally don't want to take from Zren despite upstream development slowing down. Everyone has obligations outside of volunteer work, and it would be quite rude to try and pull a "coupe". I also don't have the capacity to maintain a project of this size, there is a very real chance that this repo could go the way of upstream once the months go by.

### Integration with thunderbird or another client:
Integrate with broader client such as thunderbird. There seems to be some interest on this, but I'm unsure of how much (https://github.com/Zren/plasma-applet-eventcalendar/issues/49).
#### Pros:
- Very minor and negligible fragmentation. A new right click menu item can be added today, this can be merged upstream with minor if any issues. The actual implementation itself would be quite simple I'd imagine. I've yet to figure out how to actually make an API or xdg-open call to thunderbird, it also remains to be seem if thunderbird or other clients can actually handle such a call, shifting focus from the mail box to the calendar, and the event specifically in a deterministic manner.
#### Cons:
- Minor feature that we are currently unsure how many people would use. I would personally implement it myself and would be happy to publish it here for everyone else to use. 

### Integration with desktop environment itself:
Integrate with KDE (or another wider system feature) itself.
#### Pros:
- Probably the best "long term" option as with DE integration other apps such as thunderbird could hook into the DE itself, severing as a centralised interface and distributor.
#### Cons:
- I'm not sure I know anyone that uses Kalendar, I've downloaded it and the features seem lackluster for now. If we were to go a full EC -> KDE -> Thunderbird route, the feature set that would have be implemented could mean months if not years of work and testing.

I would love to hear everyone else's perspective on this, I think with the amount of open PRs upstream there is no lack of willing and highly capable volunteers for the future of EC. We just need to band together as community and get it done!

Thanks for taking the time to read this long post. Have a great day everyone :)

**Comment by ALikesToCode** [comment]
The main reason I am keeping this fork running and updated is to have updates on this applet for future, since the original repo seems inactive to me. I am not planning to merge this with the upstream via a PR request. My current plan and hope is to get this working on Plasma 6. 

Due to my busy schedule with work and classes, my commits might slow down. I don't want to commit to adding new features personally but will support anyone who wants to volunteer. I prefer this to be a community effort rather than a solo project, to avoid being the single point of failure in case I am unable to work on it.

Regarding improving event management, I hadn't considered it in depth. However, if you are logged in with your Google account, updates should sync even with a third-party app.

Currently, my main focus is on getting this to work on Plasma 6. You can check the progress [here](https://github.com/ALikesToCode/plasma-applet-eventcalendar/tree/plasma-6). If anyone wants to contribute, please go ahead. We can have a discussion and try to improve it together.

**Reply by yuannan** [reply]
You make a good point about Google calendar being synced. It's more so that when I get a notification from EC i sometimes want to interact with it. However, this usually opens up the webpage which is slow and breaks up the UX which I don't like.

Do you have any thoughts on features regarding events management? It would be good to understand what other users like before I plan a sprint. I understand that plama 6 is probably higher priority, but I don't use it so I don't really mind 😂.
